contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IUmbra {
+  function sendEth(
+    address payable receiver,
+    uint256 tollCommitment,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+
+  function sendToken(
+    address receiver,
+    address tokenAddr,
+    uint256 amount,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+}
+
+contract UmbraBatchSend {
+  IUmbra internal constant umbra = IUmbra(0xFb2dc580Eed955B528407b4d36FfaFe3da685401);
```

Comment:
Even though we're aiming for the same address on all networks, I think I'd still lean toward an `immutable` that is set in the constructor for more flexibility in future unexpected contexts.